fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java

```diff
@@ -0,0 +1,274 @@
+        for (Backend be : Catalog.getCurrentSystemInfo().getIdToBackend().values()) {
+            if (be.isAlive()) {
+                hostToBeId.put(be.getHost(), be.getId());
+            }
+        }
+        if (hostToBeId.isEmpty()) {
+            throw new UserException("there is no scanNode Backend");
+        }
+    }
+
+    /**
+     * Extracts predicates from conjuncts that can be pushed down to Iceberg.
+     *
+     * Since Iceberg will filter data files by metadata instead of scan data files,
+     * we pushdown all predicates to Iceberg to get the minimum data files to scan.
+     * Here are three cases for predicate pushdown:
+     * 1.The column is not part of any Iceberg partition expression
+     * 2.The column is part of all partition keys without any transformation (i.e. IDENTITY)
+     * 3.The column is part of all partition keys with transformation (i.e. MONTH/DAY/HOUR)
+     * We can use case 1 and 3 to filter data files, but also need to evaluate it in the
+     * scan, for case 2 we don't need to evaluate it in the scan. So we evaluate all
+     * predicates in the scan to keep consistency. More details about Iceberg scanning,
+     * please refer: https://iceberg.apache.org/spec/#scan-planning
+     */
+    private void preProcessConjuncts() {
+        for (Expr expr : conjuncts) {
+            UnboundPredicate unboundPredicate = toIcebergExpression(expr);
+            if (unboundPredicate != null) {
+                icebergPredicates.add(unboundPredicate);
+            }
+        }
+    }
+
+    @Override
+    public void finalize(Analyzer analyzer) throws UserException {
+        if (isFinalized) {
+            return;
+        }
+
+        LOG.debug("IcebergScanNode finalize. Tuple: {}", desc);
+        try {
+            getScanRangeLocations();
+        } catch (AnalysisException e) {
+            throw new UserException(e.getMessage());
+        }
+
+        // Min max tuple must be computed after analyzer.materializeSlots()
+        DescriptorTable descTbl = analyzer.getDescTbl();
+        minMaxTuple = descTbl.createTupleDescriptor();
+        PredicateUtils.computeMinMaxTupleAndConjuncts(analyzer, minMaxTuple, minMaxConjuncts, conjuncts);
+
+        computeStats(analyzer);
+        isFinalized = true;
+    }
+
+    @Override
+    public List<TScanRangeLocations> getScanRangeLocations(long maxScanRangeLength) {
+        return result;
+    }
+
+    public void getScanRangeLocations() throws UserException {
+        Optional<Snapshot> snapshot = IcebergUtil.getCurrentTableSnapshot(
+                srIcebergTable.getIcebergTable(), true);
+        if (!snapshot.isPresent()) {
+            LOG.info(String.format("Table %s has no snapshot!", srIcebergTable.getTable()));
+            return;
+        }
+        preProcessConjuncts();
+        for (FileScanTask task : IcebergUtil.getTableScan(
+                srIcebergTable.getIcebergTable(), snapshot.get(),
+                icebergPredicates, true).planFiles()) {
```

Comment:
I see those tasks are `FileScanTask`,  will starrocks provides any bin-pack algorithm to balance the splits between different parallelism ?

Reply:
Besides, I'm  thinking that we may need to introduce an extra data structure to handle the read process for iceberg v2 table because its split will contains both data file split and delete file split.

Reply:
The statistics here are now useless, I will delete these codes and submit another pr.